File: src/test/java/se/johannesdahlgren/adventofcode2021/Day3Test.java
package se.johannesdahlgren.adventofcode2021;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;

class Day3Test {

  @Test
  void part1Example() {
    Day3 day3 = new Day3("day3example.txt");
    var powerConsumption = day3.getPowerConsumption();
    assertThat(powerConsumption).isEqualTo(198);
  }

  @Test
  void part1() {
    Day3 day3 = new Day3("day3.txt");
    var powerConsumption = day3.getPowerConsumption();
    assertThat(powerConsumption).isEqualTo(2724524);
  }

  @Test
  void part2Example() {
    Day3 day3 = new Day3("day3example.txt");
    var powerConsumption = day3.getLifeSupportRating();
    assertThat(powerConsumption).isEqualTo(230);
  }

  @Test
  void part2() {
    Day3 day3 = new Day3("day3.txt");
    var powerConsumption = day3.getLifeSupportRating();
    assertThat(powerConsumption).isEqualTo(2775870);
  }
}

File: src/test/kotlin/se/johannesdahlgren/aoc22/Day3Test.kt
package se.johannesdahlgren.aoc22

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class Day3Test {

    @Test
    fun part1Example() {
        val day3 = Day3("day3Example")
        assertThat(day3.prioritiesSum()).isEqualTo(157)
    }

    @Test
    fun part1() {
        val day3 = Day3("day3")
        assertThat(day3.prioritiesSum()).isEqualTo(8176)
    }

    @Test
    fun part2Example() {
        val day3 = Day3("day3Example")
        assertThat(day3.prioritiesSumElfGroup()).isEqualTo(70)
    }

    @Test
    fun part2() {
        val day3 = Day3("day3")
        assertThat(day3.prioritiesSumElfGroup()).isEqualTo(2689)
    }
}